src/unstract/adapters/ocr/google_document_ai/src/google_document_ai.py

```diff
@@ -0,0 +1,167 @@
+                GoogleDocumentAIKey.CONTENT: base64.b64encode(
+                    file_content_in_bytes
+                ).decode("utf-8"),
+            },
+            GoogleDocumentAIKey.SKIP_HUMAN_REVIEW: True,
+            GoogleDocumentAIKey.FIELD_MASK: "text",
+        }
+
+    def __get_request_headers(self) -> dict[str, Any]:
+        credentials = Credentials.from_service_account_info(
+            self.google_service_account, scopes=Constants.CREDENTIAL_SCOPES
+        )
+        credentials.refresh(google_requests.Request())
+
+        return {
+            "Content-Type": "application/json; charset=utf-8",
+            "Authorization": f"Bearer {credentials.token}",
+        }
+
+    def __get_input_file_type_mime(self, input_file_path: str) -> str:
+        with open(input_file_path, mode="rb") as file_obj:
+            sample_contents = file_obj.read(100)
+            file_type = filetype.guess(sample_contents)
+
+        file_type_mime: str = (
+            file_type.MIME if file_type else FileType.TEXT_PLAIN
+        )
+
+        if file_type_mime not in FileType.ALLOWED_TYPES:
+            logger.error("Input file type not supported: " f"{file_type_mime}")
+
+        logger.info(f"file: `{input_file_path} [{file_type_mime}]`\n\n")
+
+        return file_type_mime
+
+    def process(
+        self, input_file_path: str, output_file_path: Optional[str] = None
+    ) -> str:
+        try:
+            file_type_mime = self.__get_input_file_type_mime(input_file_path)
+            if os.path.isfile(input_file_path):
+                with open(input_file_path, "rb") as fop:
+                    file_content_in_bytes: bytes = fop.read()
+            else:
+                raise AdapterError(f"File not found {input_file_path}")
+            processor_url = self.config.get(Constants.URL, "") + ":process"
+            headers = self.__get_request_headers()
+            data = self.__get_request_body(
+                file_type_mime=file_type_mime,
+                file_content_in_bytes=file_content_in_bytes,
+            )
+            response = requests.post(processor_url, headers=headers, json=data)
+            if response.status_code != 200:
+                logger.error(
+                    f"Error while calling Google Document AI: {response.text}"
+                )
+            response_json: dict[str, Any] = response.json()
+            result_text: str = response_json["document"]["text"]
+            if output_file_path is not None:
+                with open(output_file_path, "w", encoding="utf-8") as f:
+                    f.write(result_text)
+                    f.close()
+            return result_text
+        except Exception as e:
+            logger.error(f"Error while processing document {e}")
+            if not isinstance(e, AdapterError):
+                raise AdapterError(str(e))
+            else:
+                raise e
+        finally:
+            if fop is not None:
+                fop.close()
```

Comment:
I think fop will be closed automatically inside with

Reply:
Here it is used for reading a flle. I thought it should always be closed when open for reading.